--- a/Schweizer-Messer/python_module/cmake/add_python_export_library.cmake
+++ b/Schweizer-Messer/python_module/cmake/add_python_export_library.cmake
@@ -86,7 +86,8 @@
       list(APPEND BOOST_COMPONENTS python27)
     endif()
   else()
-    list(APPEND BOOST_COMPONENTS python38)
+    string(REGEX MATCH "^([0-9]+)[.]([0-9]+)" PYTHON_VERSION_PAIR "${PYTHONLIBS_VERSION_STRING}")
+    list(APPEND BOOST_COMPONENTS python${CMAKE_MATCH_1}${CMAKE_MATCH_2})
   endif()
   find_package(Boost REQUIRED COMPONENTS ${BOOST_COMPONENTS}) 
 
