--- a/aslam_optimizer/aslam_backend/include/aslam/backend/implementation/Cholmod.hpp
+++ b/aslam_optimizer/aslam_backend/include/aslam/backend/implementation/Cholmod.hpp
@@ -445,7 +445,7 @@
     template<typename I>
     void Cholmod<I>::getR(cholmod_sparse* A, cholmod_sparse** R) {
       cholmod_sparse* qrJ = cholmod_l_transpose(A, 1, &_cholmod);
-      SuiteSparseQR<double>(SPQR_ORDERING_FIXED, SPQR_NO_TOL, qrJ->ncol, 0,
+      SuiteSparseQR<double, int64_t>(SPQR_ORDERING_FIXED, SPQR_NO_TOL, qrJ->ncol, 0,
         qrJ, NULL, NULL, NULL, NULL, R, NULL, NULL, NULL, NULL, &_cholmod);
       SM_ASSERT_EQ(Exception, _cholmod.status, CHOLMOD_OK,
         "QR factorization failed");
--- a/aslam_incremental_calibration/incremental_calibration/include/aslam/calibration/algorithms/linalg.h
+++ b/aslam_incremental_calibration/incremental_calibration/include/aslam/calibration/algorithms/linalg.h
@@ -35,7 +35,7 @@
 typedef cholmod_dense_struct cholmod_dense;
 struct cholmod_common_struct;
 typedef cholmod_common_struct cholmod_common;
-template <typename Entry> struct SuiteSparseQR_factorization;
+#include <SuiteSparseQR.hpp>
 
 namespace aslam {
   namespace calibration {
--- a/aslam_incremental_calibration/incremental_calibration/include/aslam/calibration/core/LinearSolver.h
+++ b/aslam_incremental_calibration/incremental_calibration/include/aslam/calibration/core/LinearSolver.h
@@ -38,7 +38,7 @@
 
 #include "aslam/calibration/core/LinearSolverOptions.h"
 
-template <typename Entry> struct SuiteSparseQR_factorization;
+#include <SuiteSparseQR.hpp>
 
 namespace sm {
 
